### PR TITLE
JNG-4248 Add JSL SDK Tests for Filter 

### PR DESCRIPTION
**Pull request by gaborflorian**
<table>
<td>
  <a href="https://blackbelt.atlassian.net/browse/JNG-4248" title="JNG-4248" target="_blank">TaskJNG-4248</a>  Add JSL  SDK Tests for Filter / Page / Order
  </td></table>
  <br />

### CODE REVIEW

File: judo-runtime-core-jsl-itest/src/test/java/hu/blackbelt/judo/runtime/core/jsl/entity/FiltersTest.java [hunk 1]
```diff
@@ -214,6 +306,153 @@ public void testIntegerNumberFilter() {
...
+        assertTrue(zeroResult.isEmpty());
+
+        List<MyEntityWithOptionalFields> filteredByZeroResultString = myEntityWithOptionalFieldsDao
+                .query()
+                .filterBy("this.integerAttr > 1 and this.integerAttr < 2")
+                .execute();
+
+        assertTrue(filteredByZeroResultString.isEmpty());
+
+        // filter for undefined
```
Comment (robertcsakany) [inline judo-runtime-core-jsl-itest/src/test/java/hu/blackbelt/judo/runtime/core/jsl/entity/FiltersTest.java:328]: Add TODO and ticket number

File: judo-runtime-core-jsl-itest/src/test/java/hu/blackbelt/judo/runtime/core/jsl/entity/FiltersTest.java [hunk 2]
```diff
@@ -393,101 +706,583 @@ public void testStringFilter() {
...
-    @Test
-    @Requirement(reqs = {
-            "REQ-TYPE-001",
-            "REQ-TYPE-002",
-            "REQ-TYPE-004",
-            "REQ-TYPE-005",
+        List<MyEntityWithOptionalFields> iLikeByStringFilter = myEntityWithOptionalFieldsDao
+                .query()
+                .filterByStringAttr(StringFilter.ilike(STRING_1.toUpperCase()))
+                .filterBy("this.stringAttr!ilike('TEST')")
```
Comment (robertcsakany) [inline judo-runtime-core-jsl-itest/src/test/java/hu/blackbelt/judo/runtime/core/jsl/entity/FiltersTest.java]: The test intention is not clear

File: judo-runtime-core-jsl-itest/src/test/java/hu/blackbelt/judo/runtime/core/jsl/entity/FiltersTest.java [hunk 3]
```diff
@@ -393,101 +706,583 @@ public void testStringFilter() {
...
+                .filterBy("this.derivedStringAttr != 'test'")
+                .execute()
+                .get(0);
+
+        assertEquals(entity2.identifier(), notEqualsByStringFilterDerived.identifier());
+
+        List<MyEntityWithOptionalFields> iLikeByStringFilterDerived = myEntityWithOptionalFieldsDao
+                .query()
+                .filterByStringAttr(StringFilter.ilike(STRING_1.toUpperCase()))
+                .filterBy("this.derivedStringAttr!ilike('TEST')")
```
Comment (robertcsakany) [inline judo-runtime-core-jsl-itest/src/test/java/hu/blackbelt/judo/runtime/core/jsl/entity/FiltersTest.java]: The test intention is not clear

File: judo-runtime-core-jsl-itest/src/test/java/hu/blackbelt/judo/runtime/core/jsl/entity/FiltersTest.java [hunk 4]
```diff
@@ -393,101 +706,583 @@ public void testStringFilter() {
...
+                .query()
+                .filterBy("this.dateAttr >= `1999-09-19`")
+                .execute();
+
+        assertEquals(2, greaterOrEqualByFilterString.size());
+
+        List<MyEntityWithOptionalFields> lessOrEqualByFilterString = myEntityWithOptionalFieldsDao
+                .query()
+                .filterByDateAttr(DateFilter.lessOrEqualThan(DATE_1))
+                .filterBy("this.dateAttr <= `2022-07-11`")
```
Comment (robertcsakany) [inline judo-runtime-core-jsl-itest/src/test/java/hu/blackbelt/judo/runtime/core/jsl/entity/FiltersTest.java]: The test intention is not clear

File: judo-runtime-core-jsl-itest/src/test/java/hu/blackbelt/judo/runtime/core/jsl/entity/FiltersTest.java [hunk 5]
```diff
@@ -393,101 +706,583 @@ public void testStringFilter() {
...
+                .query()
+                .filterBy("this.derivedStringAttr > 'Another'")
+                .execute();
+
+        assertEquals(entity1.identifier(), greaterThanByStringFilterDerived.get(0).identifier());
+
+        Long greaterOrEqualThanByStringFilterDerived = myEntityWithOptionalFieldsDao
+                .query()
+                .filterByStringAttr(StringFilter.greaterOrEqualThan(STRING_2))
+                .filterBy("this.derivedStringAttr >= 'Another'")
```
Comment (robertcsakany) [inline judo-runtime-core-jsl-itest/src/test/java/hu/blackbelt/judo/runtime/core/jsl/entity/FiltersTest.java]: The test intention is not clear

File: judo-runtime-core-jsl-itest/src/test/java/hu/blackbelt/judo/runtime/core/jsl/entity/FiltersTest.java [hunk 6]
```diff
@@ -503,12 +1298,59 @@ public void testDateFilter() {
...
     public void testTimestampFilter() {
         List<MyEntityWithOptionalFields> list = myEntityWithOptionalFieldsDao.query().execute();
 
-        assertEquals(2, list.size());
+        assertEquals(3, list.size());
+
+        MyEntityWithOptionalFields equalToByStringFilter = myEntityWithOptionalFieldsDao
+                .query()
+                .filterByTimestampAttr(TimestampFilter.equalTo(TIMESTAMP_1))
+                .filterBy("this.timestampAttr == `2022-07-11T19:09:33`")
```
Comment (robertcsakany) [inline judo-runtime-core-jsl-itest/src/test/java/hu/blackbelt/judo/runtime/core/jsl/entity/FiltersTest.java]: The test intention is not clear

File: judo-runtime-core-jsl-itest/src/test/java/hu/blackbelt/judo/runtime/core/jsl/entity/FiltersTest.java [hunk 7]
```diff
@@ -503,12 +1298,59 @@ public void testDateFilter() {
...
+                .query()
+                .filterBy("this.timestampAttr >= `1999-09-19T09:09:09`")
+                .execute();
+
+        assertEquals(2, greaterOrEqualByStringFilter.size());
+
+        List<MyEntityWithOptionalFields> lessOrEqualByStringFilter = myEntityWithOptionalFieldsDao
+                .query()
+                .filterByTimestampAttr(TimestampFilter.lessOrEqualThan(TIMESTAMP_1))
+                .filterBy("this.timestampAttr <= `2022-07-11T19:09:33`")
```
Comment (robertcsakany) [inline judo-runtime-core-jsl-itest/src/test/java/hu/blackbelt/judo/runtime/core/jsl/entity/FiltersTest.java]: The test intention is not clear

File: judo-runtime-core-jsl-itest/src/test/java/hu/blackbelt/judo/runtime/core/jsl/entity/FiltersTest.java [hunk 8]
```diff
@@ -503,12 +1298,59 @@ public void testDateFilter() {
...
+                .execute();
+
+        assertEquals(2, lessOrEqualByStringFilter.size());
+
+        List<MyEntityWithOptionalFields> greaterThanByStringFilter = myEntityWithOptionalFieldsDao
+                .query()
+                .filterByTimestampAttr(TimestampFilter.greaterThan(TIMESTAMP_2))
+                .filterBy("this.timestampAttr > `1999-09-19T09:09:09`")
```
Comment (robertcsakany) [inline judo-runtime-core-jsl-itest/src/test/java/hu/blackbelt/judo/runtime/core/jsl/entity/FiltersTest.java]: The test intention is not clear

File: judo-runtime-core-jsl-itest/src/test/java/hu/blackbelt/judo/runtime/core/jsl/entity/FiltersTest.java [hunk 9]
```diff
@@ -553,6 +1395,101 @@ public void testTimestampFilter() {
                 .execute();
 
         assertEquals(1, lessThan.size());
+
+        // derived
+
+        MyEntityWithOptionalFields equalToByStringFilterDerived = myEntityWithOptionalFieldsDao
+                .query()
+                .filterByTimestampAttr(TimestampFilter.equalTo(TIMESTAMP_1))
+                .filterBy("this.derivedTimestampAttr == `2022-07-11T19:09:33`")
```
Comment (robertcsakany) [inline judo-runtime-core-jsl-itest/src/test/java/hu/blackbelt/judo/runtime/core/jsl/entity/FiltersTest.java]: The test intention is not clear

File: judo-runtime-core-jsl-itest/src/test/java/hu/blackbelt/judo/runtime/core/jsl/entity/FiltersTest.java [hunk 10]
```diff
@@ -553,6 +1395,101 @@ public void testTimestampFilter() {
...
+                .query()
+                .filterBy("this.derivedTimestampAttr >= `1999-09-19T09:09:09`")
+                .execute();
+
+        assertEquals(2, greaterOrEqualByStringFilterDerived.size());
+
+        List<MyEntityWithOptionalFields> lessOrEqualByStringFilterDerived = myEntityWithOptionalFieldsDao
+                .query()
+                .filterByDerivedTimestampAttr(TimestampFilter.lessOrEqualThan(TIMESTAMP_1))
+                .filterBy("this.derivedTimestampAttr <= `2022-07-11T19:09:33`")
```
Comment (robertcsakany) [inline judo-runtime-core-jsl-itest/src/test/java/hu/blackbelt/judo/runtime/core/jsl/entity/FiltersTest.java]: The test intention is not clear

File: judo-runtime-core-jsl-itest/src/test/java/hu/blackbelt/judo/runtime/core/jsl/entity/FiltersTest.java [hunk 11]
```diff
@@ -553,6 +1395,101 @@ public void testTimestampFilter() {
...
+                .execute();
+
+        assertEquals(2, lessOrEqualByStringFilterDerived.size());
+
+        List<MyEntityWithOptionalFields> greaterThanByStringFilterDerived = myEntityWithOptionalFieldsDao
+                .query()
+                .filterByDerivedTimestampAttr(TimestampFilter.greaterThan(TIMESTAMP_2))
+                .filterBy("this.derivedTimestampAttr > `1999-09-19T09:09:09`")
```
Comment (robertcsakany) [inline judo-runtime-core-jsl-itest/src/test/java/hu/blackbelt/judo/runtime/core/jsl/entity/FiltersTest.java]: The test intention is not clear

File: judo-runtime-core-jsl-itest/src/test/java/hu/blackbelt/judo/runtime/core/jsl/entity/FiltersTest.java [hunk 12]
```diff
@@ -393,101 +707,579 @@ public void testStringFilter() {
...
+            "REQ-ENT-008"
     })
-    public void testBooleanFilter() {
+    public void testStringFilter() {
         List<MyEntityWithOptionalFields> list = myEntityWithOptionalFieldsDao.query().execute();
 
-        assertEquals(2, list.size());
+        assertEquals(3, list.size());
+
+        // costume filter
```
Comment (bencegelei) [inline judo-runtime-core-jsl-itest/src/test/java/hu/blackbelt/judo/runtime/core/jsl/entity/FiltersTest.java]: costume 💃 or custom ?

**Review by bencegelei — APPROVED** [review]

**Review by robertcsakany — CHANGES_REQUESTED** [review]

File: judo-runtime-core-jsl-itest/src/test/java/hu/blackbelt/judo/runtime/core/jsl/entity/FiltersTest.java (hunk 2 above)
Comment (gaborflorian) [reply judo-runtime-core-jsl-itest/src/test/java/hu/blackbelt/judo/runtime/core/jsl/entity/FiltersTest.java]: [84f9fe7](https://github.com/BlackBeltTechnology/judo-runtime-core-jsl/pull/215/commits/84f9fe72b4ee3131dbe69c0c6cdfa8153053415a)

File: judo-runtime-core-jsl-itest/src/test/java/hu/blackbelt/judo/runtime/core/jsl/entity/FiltersTest.java (hunk 3 above)
Comment (gaborflorian) [reply judo-runtime-core-jsl-itest/src/test/java/hu/blackbelt/judo/runtime/core/jsl/entity/FiltersTest.java]: [84f9fe7](https://github.com/BlackBeltTechnology/judo-runtime-core-jsl/pull/215/commits/84f9fe72b4ee3131dbe69c0c6cdfa8153053415a)

File: judo-runtime-core-jsl-itest/src/test/java/hu/blackbelt/judo/runtime/core/jsl/entity/FiltersTest.java (hunk 4 above)
Comment (gaborflorian) [reply judo-runtime-core-jsl-itest/src/test/java/hu/blackbelt/judo/runtime/core/jsl/entity/FiltersTest.java]: [84f9fe7](https://github.com/BlackBeltTechnology/judo-runtime-core-jsl/pull/215/commits/84f9fe72b4ee3131dbe69c0c6cdfa8153053415a)

File: judo-runtime-core-jsl-itest/src/test/java/hu/blackbelt/judo/runtime/core/jsl/entity/FiltersTest.java (hunk 5 above)
Comment (gaborflorian) [reply judo-runtime-core-jsl-itest/src/test/java/hu/blackbelt/judo/runtime/core/jsl/entity/FiltersTest.java]: [84f9fe7](https://github.com/BlackBeltTechnology/judo-runtime-core-jsl/pull/215/commits/84f9fe72b4ee3131dbe69c0c6cdfa8153053415a)

File: judo-runtime-core-jsl-itest/src/test/java/hu/blackbelt/judo/runtime/core/jsl/entity/FiltersTest.java (hunk 11 above)
Comment (gaborflorian) [reply judo-runtime-core-jsl-itest/src/test/java/hu/blackbelt/judo/runtime/core/jsl/entity/FiltersTest.java]: [84f9fe7](https://github.com/BlackBeltTechnology/judo-runtime-core-jsl/pull/215/commits/84f9fe72b4ee3131dbe69c0c6cdfa8153053415a)

File: judo-runtime-core-jsl-itest/src/test/java/hu/blackbelt/judo/runtime/core/jsl/entity/FiltersTest.java (hunk 7 above)
Comment (gaborflorian) [reply judo-runtime-core-jsl-itest/src/test/java/hu/blackbelt/judo/runtime/core/jsl/entity/FiltersTest.java]: [84f9fe7](https://github.com/BlackBeltTechnology/judo-runtime-core-jsl/pull/215/commits/84f9fe72b4ee3131dbe69c0c6cdfa8153053415a)

File: judo-runtime-core-jsl-itest/src/test/java/hu/blackbelt/judo/runtime/core/jsl/entity/FiltersTest.java (hunk 1 above)
Comment (gaborflorian) [reply judo-runtime-core-jsl-itest/src/test/java/hu/blackbelt/judo/runtime/core/jsl/entity/FiltersTest.java:328]: Added

**Review by noherczeg — APPROVED** [review]

**Review by robertcsakany — APPROVED** [review]

**Pull request merged**